core/src/main/java/org/teavm/backend/lowlevel/generate/NameProviderWithSpecialNames.java

```diff
@@ -47,12 +49,14 @@ public String forMethod(MethodReference method) {
 
     private String getSpecialName(MethodReference methodReference) {
         MethodReader method = classSource.resolve(methodReference);
+        ClassReader cls = classSource.get(method.getReference().getClassName());
```

Comment:
I would not spoil compiler core with Scala-specific hacks. There's dedicated class for this called `ScalaHacks`. You can just detect these synthetic Scala classes there and remove `@Export` annotations from their methods.

Reply:
I've got it. I will redo this solution. Thanks!

Reply:
I've update the solution. Now it uses the ScalaHacks.